cirq/protocols/resolve_parameters_test.py

```diff
@@ -94,11 +109,37 @@ def _resolve_parameters_(self, resolver: ParamResolver):
     assert cirq.parameter_names(1j) == set()
 
 
-def test_skips_empty_resolution():
+@pytest.mark.parametrize(
+    'resolve_fn',
+    [
+        cirq.resolve_parameters,
+        cirq.resolve_parameters_once,
+    ],
+)
+def test_skips_empty_resolution(resolve_fn):
     class Tester:
-        def _resolve_parameters_(self, param_resolver):
+        def _resolve_parameters_(self, param_resolver, recursive):
             return 5
 
     t = Tester()
-    assert cirq.resolve_parameters(t, {}) is t
-    assert cirq.resolve_parameters(t, {'x': 2}) == 5
+    assert resolve_fn(t, {}) is t
+    assert resolve_fn(t, {'x': 2}) == 5
+
+
+def test_resolve_loops():
```

Comment:
better name: `test_recursive_resolve` (since only the last resolver has an actual loop)